src/sidebar/reducers/test/frames-test.js

```diff
@@ -41,4 +42,53 @@ describe('frames reducer', function () {
       assert.deepEqual(frames.frames(updatedState), [frame]);
     });
   });
+
+  describe('#searchUris', function () {
+    unroll('returns the expected search URIs (#when)', function (testCase) {
+      var state = init();
+      testCase.frames.forEach(function (frame) {
+        state = update(state, actions.connectFrame(frame));
+      });
+      assert.deepEqual(frames.searchUris(state), testCase.searchUris);
+    },[{
```

Comment:
I think what you propose would be cleaner in this context. I'm not too worried about readers having to understand what `unroll()` does because we use it quite extensively in the client and readers only have to read the docs once.

In cases where there is one test that you want to run with a lot of different parameters, I think it is nicer to use a style where the data comes after the test description as is the case with `unroll` and `@pytest.parametrize` in Python. In cases like this where there are a bunch of separate tests that happen to have the same setup and assert, perhaps not so much.